src/detection_win.cpp

```diff
@@ -203,49 +214,21 @@ void EIO_Find(uv_work_t* req) {
 /**********************************
  * Local Functions
  **********************************/
-void ToUpper(char * buf) {
-	char* c = buf;
-	while (*c != '\0') {
-		*c = toupper((unsigned char)*c);
-		c++;
-	}
-}
-
-
-void extractVidPid(char * buf, ListResultItem_t * item) {
-	if(buf == NULL) {
-		return;
-	}
-
-	ToUpper(buf);
-
-	char* string;
-	char* temp;
-	char* pidStr, *vidStr;
-	int vid = 0;
-	int pid = 0;
 
-	string = new char[strlen(buf) + 1];
-	memcpy(string, buf, strlen(buf) + 1);
-
-	vidStr = strstr(string, VID_TAG);
-	pidStr = strstr(string, PID_TAG);
-
-	if(vidStr != NULL) {
-		temp = (char*) (vidStr + strlen(VID_TAG));
-		temp[4] = '\0';
-		vid = strtol (temp, NULL, 16);
-	}
+void ExtractDetails(TCHAR* devicePath, ListResultItem_t* item) {
+	// expected format:
+	// \\?\USB#Vid_04e8&Pid_503b#0002F9A9828E0F06#{a5dcbf10-6530-11d2-901f-00c04fb951ed}
+	CString dpcs = devicePath + 4;
 
-	if(pidStr != NULL) {
-		temp = (char*) (pidStr + strlen(PID_TAG));
-		temp[4] = '\0';
-		pid = strtol (temp, NULL, 16);
-	}
-	item->vendorId = vid;
-	item->productId = pid;
+	CString temp = devicePath;
+	temp = temp.Left(temp.ReverseFind(_T('#')));
+	temp = temp.Mid(temp.ReverseFind(_T('#')) + 1);
```

Comment:
Let's add some comments above each step here to explain what is being extracted. Also perhaps a variable named for each step to get some better context?

Reply:
I'm not sure a new variable for each step is necessary since this is intended to cut off the beginning and ending identifiers that are not needed. I agree comments should certainly be added to clarify things.